aten/src/ATen/native/mps/operations/Indexing.mm

```diff
@@ -28,6 +28,152 @@
+    // Get a list of offsets to index the Input, Output and the Index tensors
+    simd_uint3* data_offsets = (simd_uint3*)[offsets contents];
+    getDataOffsets(iter, data_offsets, iter_numel);
+
+    id<MTLCommandBuffer> commandBuffer = mpsStream->commandBuffer();
+    id<MTLComputeCommandEncoder> computeEncoder = [commandBuffer computeCommandEncoder];
+
+    MTLFunctionConstantValues* constantValues = [MTLFunctionConstantValues new];
+    int64_t storage_offset = inputTensor.storage_offset();
+    [constantValues setConstantValue: &storage_offset type:MTLDataTypeLong atIndex:0];
+    [constantValues setConstantValue: &num_indices type:MTLDataTypeUInt atIndex:1];
+
+    id<MTLFunction> indexKernelFunction = MPSDevice::getInstance()->metalFunction(indexFunction, constantValues);
+    id<MTLArgumentEncoder> argumentEncoder = [indexKernelFunction newArgumentEncoderWithBufferIndex:0];
+    NSUInteger argumentBufferLength = argumentEncoder.encodedLength;
+    id<MTLBuffer> indexAB = [device newBufferWithLength:argumentBufferLength options:0];
+    [argumentEncoder setArgumentBuffer:indexAB offset:0];
+
+    for (uint32_t idx = 0; idx < num_indices; idx++) {
+      const Tensor& indexTensor = iter.tensor(idx+2);
+      [argumentEncoder setBuffer: getMTLBufferStorage(indexTensor)
+                          offset: 0
+                         atIndex: idx];
+      TORCH_CHECK(indexTensor.scalar_type() == ScalarType::Long, "index(): Expected dtype int64 for index");
+    }
+
+    id<MTLBuffer> indexSize = [device newBufferWithBytes: index_size.data()
+                                                  length: sizeof(index_size[0]) * index_size.size()
+                                                 options: options];
+    id<MTLBuffer> indexStride = [device newBufferWithBytes: index_stride.data()
+                                                    length: sizeof(index_stride[0]) * index_stride.size()
+                                                   options: options];
+
+    // FIXME: PSO needs to be cached
+    id<MTLComputePipelineState> indexSelectPSO = [device newComputePipelineStateWithFunction: indexKernelFunction
+                                                                                       error: &error];
+    TORCH_CHECK(indexSelectPSO, "Failed to created pipeline state object, error: ", [[error description] UTF8String]);
+
+    for (uint32_t idx = 0; idx < num_indices; idx++) {
+      const Tensor& indexTensor = iter.tensor(idx+2);
+      [computeEncoder useResource:getMTLBufferStorage(indexTensor) usage:MTLResourceUsageRead];
+    }
+
+    [computeEncoder setComputePipelineState: indexSelectPSO];
+    [computeEncoder setBuffer: indexAB offset:0 atIndex:0];
+    [computeEncoder setBuffer: indexSize offset:0 atIndex:1];
+    [computeEncoder setBuffer: indexStride offset:0 atIndex:2];
+    [computeEncoder setBuffer: offsets offset:0 atIndex:3];
+    [computeEncoder setBuffer: inputBuffer offset:0 atIndex:4];
```

Comment:
```suggestion
    [computeEncoder setBuffer: inputBuffer offset:inputTensor.storage_offset() atIndex:4];
```

Reply:
Fixed in next [PR](https://github.com/kulinseth/pytorch/pull/77/)